core/src/main/java/hudson/FilePath.java

```diff
@@ -2280,8 +2284,7 @@ private void readFromTar(String name, File baseDir, InputStream in) throws IOExc
                     if (parent != null) mkdirs(parent);
                     writing(f);
 
-                    byte linkFlag = (Byte) LINKFLAG_FIELD.get(te);
-                    if (linkFlag==TarEntry.LF_SYMLINK) {
+                    if (te.isSymbolicLink()) {
```

Comment:
Most probably we're talking about different code sections:
![screen shot 2015-04-14 at 17 14 32](https://cloud.githubusercontent.com/assets/3000480/7138578/edc8762c-e2c9-11e4-9940-64a185b2d569.png)